### PR TITLE
Add gps_adjust_week_cycle for handling week rollover

### PR DESCRIPTION
**Pull request by ljbade**
First step at addressing #265 

Next step will be to make the reference week a firmware setting,.

### CODE REVIEW

**Comment by ljbade** [comment]
@fnoble I recall you had concern about hard coding `GPS_WEEK_REFERENCE`?

This is my reasoning for having it in libswiftnav:
- For a libswiftnav unit tests I needed this in here
- In the firmware this value will only be used as the compile time default
- The user will be able to change this via a setting

Until I figure out how to calculate current GPS WN at compile time, this number will have to be updated by hand for a new firmware release.

**Comment by ljbade** [comment]
@swift-nav/firmware Ready for review

File: src/time.c [hunk 1]
```diff
@@ -101,3 +101,21 @@ void gps_time_match_weeks(gps_time_t *t, const gps_time_t *ref)
...
+
+/** Adjust the week number of t to correctly refelct the current week cycle.
+ *
+ * Assumes the current week number cannot be earlier than the reference WN. So
+ * will return the correct WN for at most 20 years after the reference WN.
+ *
+ * TODO the reference value should be set from the build time, as well as being
+ * able to be changed via a flash setting.
+ */
+void gps_adjust_week_cycle(gps_time_t *t, u16 wn_ref) {
```
Comment (jacobmcnamee) [inline src/time.c]: Can we make this work regardless of whether or not t has already been adjusted?

**Comment by ljbade** [comment]
@jacobmcnamee I added a check in case t was already adjusted, can you review?

File: src/time.c (hunk 1 above)
Comment (jacobmcnamee) [inline src/time.c]: This still breaks if t->wn is less than wn_ref but not zero.

How about this:
```
if (t->wn < wn_ref)
  t->wn += 1024 * ((wn_ref + 1023 - t->wn) / 1024);
```

Comment (jacobmcnamee) [inline src/time.c]: Also as mentioned I think we should take in only the uncorrected week number and adopt the convention that gps_time_t only ever contains the corrected / absolute week number.

**Comment by ljbade** [comment]
I changed the signature of `gps_adjust_week_cycle` to prevent accidental use with a `gpstime_t` that was already corrected. This function is only intended for use in `ephemeris.c`

**Comment by jacobmcnamee** [comment]
Idk where my inline comments went, but how would you feel about changing to
```
if (wn_raw >= wn_ref)
  return wn_raw;

return wn_raw + 1024 * ((wn_ref + 1023 - wn_raw) / 1024);
```

**Comment by ljbade** [comment]
@jacobmcnamee Done

**Pull request merged**